src/python/pants/backend/docker/util_rules/docker_build_context.py

```diff
@@ -166,7 +166,8 @@ def _get_stages_and_tags(
     ) -> tuple[set[str], dict[str, str]]:
         # Go over all FROM tags and names for all stages.
         stage_names: set[str] = set()
-        stage_tags = (tag.split(maxsplit=1) for tag in dockerfile_info.version_tags)
+        # tag is empty if image is referenced by digest instead
+        stage_tags = ((tag.split(maxsplit=1) + [""])[:2] for tag in dockerfile_info.version_tags)
         tags_values: dict[str, str] = {}
         for idx, (stage, tag) in enumerate(stage_tags):
             if tag.startswith("build-arg:"):
```

Comment:
(Comment is on the closest line I can place: this actually applies to lines 193-196): I think we should make the code that sets `tags_value` below conditional on `tag` actually existing:

```python
if tag:
    if idx == 0:
        ...
    tag_values[stage] = tag
```

If not, we'll have `{..., "stage0": ""}` in the `tag_values` (used for interpolating https://www.pantsbuild.org/2.24/docs/docker/tagging-docker-images#string-interpolation-using-placeholder-values ) which doesn't seem super helpful?